### PR TITLE
feat: access TF using Authorization header

### PR DESCRIPTION
**Pull request by jamacku**
Passing api_key in data object is now deprecated.

### CODE REVIEW

**Comment by jamacku** [comment]
[test]

**Comment by jamacku** [comment]
Our tests are using the new Authorization header and passing, so I think it should be safe to merge this PR.

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "@octokit/plugin-throttling": "9.3.1",
     "@octokit/rest": "21.0.2",
     "@octokit/types": "13.5.0",
-    "testing-farm": "^1.9.2",
+    "testing-farm": "1.10.0",
```
Comment (phracek) [inline package.json:37]: I don't follow this. Sometimes we have `^1.9.2` and sometimes `1.10.0`. I would prefer to keep format.

Comment (jamacku) [reply package.json:37]: This is my fault. I changed it in the last PR and didn't notice before merging. I prefer the format without `^`.

File: src/action.ts [hunk 2]
```diff
@@ -39,7 +39,10 @@ import {
 async function action(pr: PullRequest): Promise<void> {
   const tfInstance = getInput('api_url');
 
-  const api = new TestingFarmAPI(tfInstance);
+  const api = new TestingFarmAPI(
```
Comment (phracek) [inline src/action.ts:42]: Only nit-pick. I would prefer to add there a link to testing-farm repo I would like to find usage. But not blocking this PR. WDYT.

**Review by phracek — CHANGES_REQUESTED** [review]
Thanks for this pull request. Some nit-picks. Only. Good work. I would prefer to update also README.md with reference to testing-farm repo in plumbers team, so user know where is API implementation.

File: src/action.ts (hunk 2 above)
Comment (jamacku) [reply src/action.ts:43]: Good point, done

**Comment by jamacku** [comment]
> Thanks for this pull request. Some nit-picks. Only. Good work. I would prefer to update also README.md with reference to testing-farm repo in plumbers team, so user know where is API implementation.

I have added a note about testing-farm lib to the README. PTAL

**Review by phracek — APPROVED** [review]
LGTM. Thanks for this.

**Pull request merged**